.github/workflows/clang-tidy.yaml

```diff
@@ -32,24 +30,23 @@ jobs:
   clang-tidy:
     name: Run clang-tidy
     runs-on: ubuntu-latest
+    # be consistent with tools/toolchain/image
+    container: scylladb/scylla-toolchain:fedora-40-20240621
```

Comment:
It would be nice if this were kept in sync with`tools/toolchain/image` somehow. Perhaps there is a way to include the contents of that file here?

In any case, I don't think it's a blocker for this PR.

Reply:
yeah, i thought about this. one way to address this is probably to create yet another workflow which can be called by this, and to use an input parameter for this. i will create a follow up to implement this.

Reply:
created #19655